src/pages/tutorial/cw-contract/testing-query.mdx

```diff
@@ -0,0 +1,185 @@
+# Testing a query
+
+Last time we created a new query, now it is time to test it out. We will start with the basics - the
+unit test. This approach is simple and doesn't require knowledge besides Rust. Go to the
+`src/contract.rs` and add a test in its module:
+
+```rust copy filename="src/contract.rs"
+// ...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn greet_query() {
+        let resp = query::greet().unwrap();
+        assert_eq!(
+            resp,
+            GreetResp {
+                message: "Hello World".to_owned()
+            }
+        );
+    }
+}
+```
+
+If you ever wrote a unit test in Rust, nothing should surprise you here. Just a simple test-only
+module contains local function unit tests. The problem is - this test doesn't build yet. We need to
+tweak our message types a bit.
+
+```rust {3, 8} copy filename="src/msg.rs" template="empty"
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub struct GreetResp {
+    pub message: String,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub enum QueryMsg {
+    Greet {},
+}
+```
+
+I added three new derives to both message types.
+[`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) is required to allow comparing
+types for equality - so we can check if they are equal. The
+[`Debug`](https://doc.rust-lang.org/std/fmt/trait.Debug.html) is a trait generating debug-printing
+utilities. It is used by [`assert_eq!`](https://doc.rust-lang.org/std/macro.assert_eq.html) to
+display information about mismatch if an assertion fails. Note that because we are not testing the
+`QueryMsg` in any way, the additional trait derives are optional. Still, it is a good practice to
+make all messages both `PartialEq` and `Debug` for testability and consistency. The last one,
+[`Clone`](https://doc.rust-lang.org/std/clone/trait.Clone.html) is not needed for now yet, but it is
+also good practice to allow messages to be cloned around. We will also require that later, so I
+added it already not to go back and forth.
```

Comment:
```suggestion
good practice to allow messages to be cloned around. We will also require that later.
```

Removed repetition (also). I also don't think that explanation at the end is needed ;)

Reply:
Good point ;)